src/efi.rs

```diff
@@ -506,6 +656,99 @@ impl Component for Efi {
         Ok(meta)
     }
 
+    fn extend_payload(&self, sysroot_path: &str, src_input: &str) -> Result<Option<bool>> {
+        let dest_efidir_base = Path::new(sysroot_path).join("usr/lib/efi").join("firmware");
+
+        let src_input_path = Path::new(src_input);
+        let path_to_query = if src_input_path.is_dir() {
+            WalkDir::new(src_input_path)
+                .into_iter()
+                .filter_map(|e| e.ok())
+                .find(|e| e.file_type().is_file())
+                .map(|e| e.path().to_path_buf())
+                .ok_or_else(|| anyhow!("No file found in directory {}", src_input))?
+        } else {
+            src_input_path.to_path_buf()
+        };
+
+        let meta_from_src = packagesystem::query_files(sysroot_path, [path_to_query])
+            .context(format!("Querying RPM metadata for {:?}", src_input_path))?;
+
+        let version_string_part =
+            meta_from_src.version.split(',').next().ok_or_else(|| {
+                anyhow!("RPM query returned an empty or malformed version string")
+            })?;
+
+        let parts: Vec<&str> = version_string_part.split('-').collect();
```

Comment:
Hmm @HuijingHei is reworking the version parsing in https://github.com/coreos/bootupd/pull/977 etc.